keras/callbacks.py

```diff
@@ -655,9 +661,29 @@ def __init__(self, log_dir='./logs',
         self.embeddings_metadata = embeddings_metadata or {}
         self.batch_size = batch_size
 
+        if K.backend() != 'tensorflow':
+            compatibility_dict = {"histogram_freq": 0,
+                                  "write_graph": False,
+                                  "write_images": False,
+                                  "embeddings_freq": 0}
+
+            warning_message = ["You're not using TensorFlow as backend. ",
+                               "Certain functions of the TensorBoard callback are going to be disabled.\n"]
```

Comment:
I don't think the warning message ends up being very clear... Better to raise specific warnings below, one at a time